app/javascript/mastodon/main.js

```diff
@@ -27,6 +27,12 @@ function main() {
   // import customization styles
   require.context('../../assets/stylesheets/', false, /custom.*\.scss$/);
 
+  if (process.env.NODE_ENV === 'production') {
+    // avoid offline in dev mode because it's harder to debug
+    const OfflinePluginRuntime = require('offline-plugin/runtime');
+    OfflinePluginRuntime.install();
```

Comment:
> Do you mean ServiceWorker blocks loading of other resources?

@NekR Don't know how it would interact with the main thread in the wild (grr.. grr... previous versions of Chrome). So, it should be safe to use it after the load event.

> Even if that's the case, I think `requestIdleCallback()` would be a better choice.

@nolanlawson `requestIdleCallback()` can be used to do like less important stuff (error reporting, analytics and stuff). In some cases it might not even be called (like on a slow mobile device).

There is still we can add from [https://youtu.be/CPP9ew4Co0M?t=12m2s](https://youtu.be/CPP9ew4Co0M?t=12m2s) 😜